### PR TITLE
Add verbose flag

### PR DESCRIPTION
**Pull request by john-sandall**
Initial sketch of a solution to #153 

These are my changes so far in their entirety (within `environment.py::Environment::create_lockfile()`):

```python
verbose = True
process = subprocess.Popen(
    self.pin_command,
    stdout=subprocess.PIPE,
    stderr=None if verbose else subprocess.PIPE,
)
if verbose:
    while True:
        output = process.stdout.readline()
        if output == b'' and process.poll() is not None:
            break
        if output:
            print(output.strip())
    # TODO: Decide what to do about this
    # stdout, stderr = process.poll(), None
```

I haven't add the CLI argument for `verbose` as I'm not sure of the best way to hook this up;  also I'm not sure if this solution is workable. I've tested it, and it works, but there may be drawbacks from doing this that I'm not aware of and I'm pretty sure that using `print(output.strip())` here isn't correct (should it be `logger.info()` or `sys.stdout.write()` maybe?).

Hopefully it's a starting point though?

It's passing `flake8` and I've run pytest:

```
ERROR    pip-compile-multi:environment.py:218 Package django was resolved to different versions in different environments: 1.10.8 and 3.1.7
PASSED
tests/test_discover.py::test_discover_nested PASSED
tests/test_pipcompilemulti.py::test_fix_compatible_pin PASSED
tests/test_pipcompilemulti.py::test_no_fix_incompatible_pin PASSED
tests/test_pipcompilemulti.py::test_pin_is_ommitted_if_set_to_ignore PASSED
tests/test_pipcompilemulti.py::test_post_releases_are_kept_by_default PASSED
tests/test_pipcompilemulti.py::test_forbid_post_releases PASSED
tests/test_pipcompilemulti.py::test_parse_references[base.in-refs0] PASSED
tests/test_pipcompilemulti.py::test_parse_references[test.in-refs1] PASSED
tests/test_pipcompilemulti.py::test_parse_references[local.in-refs2] PASSED
tests/test_pipcompilemulti.py::test_split_header PASSED
tests/test_pipcompilemulti.py::test_concatenation PASSED
tests/test_pipcompilemulti.py::test_parse_hashes_with_comment PASSED
tests/test_pipcompilemulti.py::test_parse_hashes_without_comment PASSED
tests/test_pipcompilemulti.py::test_serialize_hashes PASSED
tests/test_pipcompilemulti.py::test_reference_cluster PASSED
tests/test_pipcompilemulti.py::test_parse_vcs_dependencies PASSED
tests/test_pipcompilemulti.py::test_merged_packages_raise_for_conflict
-------------------------------------------------------------------------------------------------------------- live log call ---------------------------------------------------------------------------------------------------------------
ERROR    pip-compile-multi:utils.py:79 Package x was resolved to different versions in different environments: 2 and 1
PASSED
tests/test_pipcompilemulti.py::test_fix_pin_detects_version_conflict
-------------------------------------------------------------------------------------------------------------- live log call ---------------------------------------------------------------------------------------------------------------
ERROR    pip-compile-multi:environment.py:218 Package x was resolved to different versions in different environments: 2 and 1
PASSED
tests/test_upgrade_feature.py::test_upgrade_package_disables_upgrade PASSED
tests/test_utils.py::test_recursive_refs PASSED

====================================================================================================== 49 passed in 114.33s (0:01:54) ======================================================================================================
```

### CODE REVIEW

File: pipcompilemulti/environment.py [hunk 1]
```diff
@@ -63,11 +63,23 @@ def create_lockfile(self):
             if sink_out_path and sink_out_path != self.outfile:
                 original_in_file = self._read_infile()
                 self._inject_sink()
+
+            verbose = True
             process = subprocess.Popen(
                 self.pin_command,
                 stdout=subprocess.PIPE,
```
Comment (graingert) [inline pipcompilemulti/environment.py:70]: I think the stdout is captured only for logging it later, so we could just set: `stdout=None` and then skip logging stdout in verbose mode

Comment (john-sandall) [reply pipcompilemulti/environment.py:70]: Setting the `stdout=None` here gives:

```
Traceback (most recent call last):
  File "/Users/john/.virtualenvs/pip-compile-multi-demo/lib/python3.8/site-packages/pipcompilemulti/cli_v1.py", line 26, in cli
    recompile()
  File "/Users/john/.virtualenvs/pip-compile-multi-demo/lib/python3.8/site-packages/pipcompilemulti/actions.py", line 30, in recompile
    compile_topologically(env_confs, deduplicator)
  File "/Users/john/.virtualenvs/pip-compile-multi-demo/lib/python3.8/site-packages/pipcompilemulti/actions.py", line 39, in compile_topologically
    if env.maybe_create_lockfile():
  File "/Users/john/.virtualenvs/pip-compile-multi-demo/lib/python3.8/site-packages/pipcompilemulti/environment.py", line 50, in maybe_create_lockfile
    self.create_lockfile()
  File "/Users/john/.virtualenvs/pip-compile-multi-demo/lib/python3.8/site-packages/pipcompilemulti/environment.py", line 74, in create_lockfile
    output = process.stdout.readline()
AttributeError: 'NoneType' object has no attribute 'readline'
```

Comment (graingert) [reply pipcompilemulti/environment.py:70]: Yeah you won't need to use process.stdout anymore as it should be sent to the terminal already

Comment (john-sandall) [reply pipcompilemulti/environment.py]: Oh my that makes it simpler

**Comment by peterdemin** [comment]
Looking good. The outstanding steps:
* Add `-v/--verbose` option through a feature in `features/` directory.
* Fix logging for pip-tools failure.

**Comment by peterdemin** [comment]
Closing in favor of #251

**Pull request closed without merging**